src/main/java/menion/android/whereyougo/network/LoginTask.java

```diff
@@ -0,0 +1,237 @@
+
+    private final String username;
+    private final String password;
+    private final AlertDialog loginCheckdialog;
+    private OkHttpClient httpClient;
+    private String errorMessage;
+
+    public LoginTask(View view, String username, String password) {
+        super();
+        this.username = username;
+        this.password = password;
+
+        // Inflate Custom alert dialog view
+        View customAlertDialogView = LayoutInflater.from(view.getContext())
+            .inflate(R.layout.custom_progress_dialog, null, false);
+        loginCheckdialog = new MaterialAlertDialogBuilder(view.getContext())
+            .setView(customAlertDialogView)
+            .setTitle("Testing the Login Credentials")
+            .setMessage("Working")
+            .setNegativeButton("Cancel Check", (dialog, which) -> {
+                this.cancel(true);
+            })
+            .create();
+    }
+
+    @Override
+    protected void onPreExecute() {
+        super.onPreExecute();
+        loginCheckdialog.show();
+    }
+
+    @Override
+    protected Boolean doInBackground(Void... params) {
+        return init() && ping() && login() && logout();
+    }
+
+    @Override
+    protected void onProgressUpdate(Progress... values) {
+        super.onProgressUpdate(values);
+        switch (values[0].state) {
+            case FAIL:
+                loginCheckdialog.setMessage("Logincheck not successful");
+                loginCheckdialog.getButton(AlertDialog.BUTTON_NEGATIVE).setText("Close Dialog");
+                loginCheckdialog.findViewById(R.id.dialogProgressBarLayout).setVisibility(View.GONE);
+                break;
+            case SUCCESS:
+                loginCheckdialog.setMessage("Logincheck successful");
+                loginCheckdialog.getButton(AlertDialog.BUTTON_NEGATIVE).setText("Close Dialog");
+                loginCheckdialog.findViewById(R.id.dialogProgressBarLayout).setVisibility(View.GONE);
+                break;
+            case WORKING:
+                switch (values[0].task) {
+                    case INIT:
+                        loginCheckdialog.setMessage("Init");
+                        break;
+                    case PING:
+                        loginCheckdialog.setMessage("Pinging");
+                        break;
+                    case LOGIN:
+                        loginCheckdialog.setMessage("Logging in");
+                        break;
+                    case LOGOUT:
+                        loginCheckdialog.setMessage("Logging out");
+                        break;
+                    default:
+                        loginCheckdialog.setMessage("Something unexpected happened. Please close the dialog!");
+                        break;
+                }
+                break;
+            default:
+                loginCheckdialog.setMessage("Something unexpected happened. Please close the dialog!");
```

Comment:
In case for example that we receive a network disconnect or a networked changed event, the Task can't finish and the network possibly will throw a stacktrace, thus the error message for the user. Since we define the well known states inside the `case`, we want to catch all other situations to be safe on the usability side.

Reply:
In the error situation you set the Button text to pref_gc_check_dialog_negative_button_close, the same as in the success situation.
The text "Please close the dialog!" is is therefore superfluous.